plugin/backend_lookup.go

```diff
@@ -389,9 +389,17 @@ func NS(b ServiceBackend, zone string, state request.Request, opt Options) (reco
 	return records, extra, nil
 }
 
+func min(a, b uint32) uint32 {
+	if a < b {
+		return a
+	}
+	return b
+}
```

Comment:
I didn't know where to put this function, so I just dropped it here. Assuming you're in favor of this PR, let me know if there's a better place for this.

Reply:
Looks simple enough to me to just do that without a function.

```go
ttl, minTTL := 300, b.MinTTL(state)
if ttl > minTTL {
    ttl = minTTL
}
```